app/helpers/form_helper.rb

```diff
@@ -198,17 +198,14 @@ def file_field_f(f, attr, options = {})
   end
 
   def autocomplete_f(f, attr, options = {})
-    field(f, attr, options) do
-      path = options.delete(:path) || send("#{f.object.class.pluralize.underscore}_path") if options[:full_path].nil?
-      auto_complete_search(attr,
```

Comment:
I think it still exist in some plugins,
need to be careful with it, maybe throw a warning whenever this is being used?
also I don't think it is in this PR scope.

Reply:
it may be used in other plugins, which is why i suggested deprecating it rather than completely removing. it can be a separate pr, but if we don't take care of it now it will just remain as dead code.

Reply:
I agree, what is our best practices to deprecate methods on the server side?

Reply:
Add `Foreman::Deprecation.deprecation_warning(foreman_version_deadline, message)` in the function.

Reply:
Looks like it is used by REX, but it would still make sense to deprecate and update rex to also use the react component

Reply:
fixed